google-datacatalog-qlik-connector/src/google/datacatalog_connectors/qlik/scrape/websocket_responses_manager.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+
+
+class WebsocketResponsesManager:
+    """Utility class with common features for responses handling over websocket
+    communication sessions."""
+
+    def __init__(self):
+        self.__pending_response_ids = {}
+        self.__unhandled_responses = []
+        # Used by the consumer to notify the producer on new responses, so the
+        # producer can take actions such as sending follow up requests.
+        self.__new_response_event = asyncio.Event()
+
+    def add_pending_ids_list(self, key):
```

Comment:
The name of this method does not seem to be coherent with the logic. This is initializing the key, perhaps `add_pending_ids_list` -> `init_pending_response_ids_key`  ?

Reply:
I'm ok with either `add` or `init`, but I understand the method initializes a list using the provided key to reference it later. It doesn't initialize the key itself... WDYT?

Reply:
If you are ok with it, I think `init` is more clear, `add`  gives the false impression that the key is being added, which is not true.

Reply:
Do you mean `init_pending_ids_list`?

Reply:
yea, I'm ok with either `init_pending_ids_list` or `init_pending_ids` .

Because `__pending_response_ids` is a dict, and the way I see it, the code is initializing the key with a list.

But the list part seems to be an internal implementation detail, users of this class need to be aware that a list is being used? Otherwise I would just take list away from the method name, let's say you want change list to other data structure like a set or something else, users of this class need to be aware of this?

Reply:
I'm ok with `init_pending_ids`. Tks!

Reply:
Sorry @mesmacosta, I'm performing the changes here and by `init_pending_ids` a user might think we are initializing the ids, and not the data structure that will store the pending ids over the communication session. So my suggestion is to rename to `init_pending_ids_holder` and `init_pending_ids_holders`. Are you ok with these names?